Analysis/interface/GenLepton.h

```diff
@@ -95,67 +98,70 @@ class GenLepton {
         return leptons;
     }
 
-    template<typename IntVector, typename LongVector, typename FloatVector>
-    static GenLepton fromRootTuple(int lastMotherIndex,
-                                   const IntVector& genParticle_pdgId,
-                                   const LongVector& genParticle_mother,
-                                   const IntVector& genParticle_charge,
-                                   const IntVector& genParticle_isFirstCopy,
-                                   const IntVector& genParticle_isLastCopy,
-                                   const FloatVector& genParticle_pt,
-                                   const FloatVector& genParticle_eta,
-                                   const FloatVector& genParticle_phi,
-                                   const FloatVector& genParticle_mass,
-                                   const FloatVector& genParticle_vtx_x,
-                                   const FloatVector& genParticle_vtx_y,
-                                   const FloatVector& genParticle_vtx_z)
+    template <template <typename> class CollectionType>
+    static GenLepton fromRootTuple(
+                                   const bool isGeantPlusSim_,
```

Comment:
- Have you checked that this template syntax well supported by gcc7 (to be compatible with UL code)?
- for compatibility with default gen particle representation please, keep function arguments as before and add
`std::optional<CollectionType<int>> fromGeant = std::optional<CollectionType<int>>`